agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -201,6 +201,7 @@ def start(info)
         end
       end
 
+      attach_router unless interface_ip('weave')
```

Comment:
Yes. I tried 20+ reboots with this PR and in all of those everything worked as expected.